src/spi/blocking.rs

```diff
@@ -4,15 +4,41 @@
 //! traits. To save boilerplate when that's the case a `Default` marker trait may be provided.
 //! Implementing that marker trait will opt in your type into a blanket implementation.
 
-/// Blocking transfer
+/// Blocking transfer with separate buffers
 pub trait Transfer<W> {
     /// Error type
     type Error: core::fmt::Debug;
 
+    /// Writes and reads simultaneously. `write` is written to the slave on MOSI and
+    /// words received on MISO are stored in `read`.
+    ///
+    /// It is allowed for `read` and `write` to have different lengths, even zero length.
+    /// The transfer runs for `max(read.len(), write.len())` words. If `read` is shorter,
+    /// incoming words after `read` has been filled will be discarded. If `write` is shorter,
+    /// the value of words sent in MOSI after all `write` has been sent is implementation-defined,
+    /// typically `0x00`, `0xFF`, or configurable.
```

Comment:
has this always been the case or have we implicitly been assuming `read.len() == write.len()`? and is this sensible to represent for more complex implementors (ie. using DMA, linux)?

we have `Transactional` for building sequences, if you need to do varying length things, it would seem to me to be simpler (or maybe, [less surprising](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) to demand the same lengths here?

Reply:
This was [discussed](https://matrix.to/#/!BHcierreUuwCMxVqOf:matrix.org/$ONZJiK_kqPu2beGIGMicS4e1XWNVhOrxE1P_C5gZQEs?via=matrix.org&via=psion.agg.io&via=mozilla.org) in today's REWG meeting. TLDR: seems people are OK with differing lengths.

Also the possibility of allowing impls to support it or not was discussed, but IMO it defeats the point since drivers wouldn't be able to use it.